jupyter/amd/pytorch/ubi9-python-3.9/Dockerfile

```diff
@@ -0,0 +1,102 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+LABEL name="odh-notebook-rocm-python-3.9" \
+      summary="ROCm Python 3.9 base image for ODH notebooks" \
+      description="ROCm Python 3.9 builder image based on CentOS Stream 9 for ODH notebooks" \
+      io.k8s.display-name="ROCm Python 3.9 base image for ODH notebooks" \
+      io.k8s.description="ROCm Python 3.9 builder image based on C9S for ODH notebooks" \
+      authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+      io.openshift.build.commit.ref="main" \
+      io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/amd/python-3.9" \
+      io.openshift.build.image="quay.io/opendatahub/workbench-images:rocm-python-3.9"
+
+USER 0
+WORKDIR /opt/app-root/bin
+
+ARG ROCM_VERSION=6.1
+ARG AMDGPU_VERSION=6.1
+# Base
+RUN yum -y install git java-1.8.0-openjdk python; yum clean all
+# Install ROCm AMD from:
+# https://github.com/ROCm/ROCm-docker/blob/master/dev/Dockerfile-centos-7-complete
+# Enable epel-release repositories
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    yum config-manager --set-enabled crb && \
+    yum clean all && rm -rf /var/cache/yum
+
+
+# Install required base build and packaging commands for ROCm
+RUN yum -y install \
+    bc \
+    bridge-utils \
+    glibc.i686 \
+    numactl-libs \
+    libssh \
+    libunwind-devel \
+    libunwind \
+    pciutils \
+    pciutils-libs \
+    re2c \
+    doxygen \
+    elfutils-libelf-devel \
+    expect \
+    numactl-devel \
+    pciutils-devel \
+    qemu-kvm \
+    subversion \
+    dkms \
+    dpkg \
+    dpkg-dev \
+    dpkg-perl && \
+    yum clean all 
+# These binaries are not available.
+# devscripts \
+# pth \
+# libcxx-devel \
+
+
+# Enable the epel repository for fakeroot
+# RUN yum --enablerepo=extras install -y fakeroot
+# RUN yum clean all
+RUN yum install -y fakeroot && \
+    yum clean all
+
+
+# On CentOS, install package centos-release-scl available in CentOS repository:
+# RUN yum install -y centos-release-scl
+
+# Install the devtoolset-7 collection:
+# RUN yum install -y devtoolset-7
+# RUN yum install -y devtoolset-7-libatomic-devel devtoolset-7-elfutils-libelf-devel
+
+# Install the ROCm rpms
+RUN echo "[ROCm]" > /etc/yum.repos.d/rocm.repo && \
+    echo "name=ROCm" >> /etc/yum.repos.d/rocm.repo && \
+    echo "baseurl=https://repo.radeon.com/rocm/el9/$ROCM_VERSION/main" >> /etc/yum.repos.d/rocm.repo && \
+    echo "enabled=1" >> /etc/yum.repos.d/rocm.repo && \
+    echo "gpgcheck=0" >> /etc/yum.repos.d/rocm.repo
+
+RUN echo "[amdgpu]" > /etc/yum.repos.d/amdgpu.repo && \
+    echo "name=amdgpu" >> /etc/yum.repos.d/amdgpu.repo && \
+    echo "baseurl=https://repo.radeon.com/amdgpu/$AMDGPU_VERSION/el/9.2/main/x86_64" >> /etc/yum.repos.d/amdgpu.repo && \
```

Comment:
Moreover - we use [this image](https://github.com/opendatahub-io/notebooks/blob/main/base/ubi9-python-3.9/Dockerfile#L1) as the very base for our UBI9 images:
* https://catalog.redhat.com/software/containers/ubi9/python-39/61a61032bfd4a5234d59629e?architecture=amd64&image=666134175ea3c2839df25dad&container-tabs=dockerfile

That image states that it's based on this image:
```
FROM ubi9/s2i-base:rhel9.4.0
```

So instead of the `9.2` version, we should use `9.4`, thus something like this:
```
ARG EL_VERSION=9.4
```

We can either have this hardcoded as above (disadvantage that we need to think about this and update it when necessary), or we can call something like this eventually:
```
cat /etc/os-release | grep "VERSION_ID" | sed 's#.*"\(.*\)"#\1#'
```
Disadvantage in this second case is that we may possibly hit a window when the AMD sources won't be available for the particular UBI version yet (e.g. since it's too fresh - aka we updated our base image too soon).

From my point of view it's a bit better to use the latter approach since this gives as an heads up at least. But I don't insist.

Reply:
Marking this a resolved, as wrong Dockerfile was pushed earlier.